docs/machine-learning/how-to-guides/train-machine-learning-model-ml-net.md

```diff
@@ -0,0 +1,179 @@
+
+```csharp
+public class HousingData
+{
+    [LoadColumn(0)]
+    public float Size { get; set; }
+
+    [LoadColumn(1, 3)]
+    [VectorType(3)]
+    public float[] HistoricalPrices { get; set; }
+
+    [LoadColumn(4)]
+    [ColumnName("Label")]
+    public float CurrentPrice { get; set; }
+}
+```
+
+Load the data into an [`IDataView`](xref:Microsoft.ML.IDataView):
+
+```csharp
+HousingData[] housingData = new HousingData[]
+{
+    new HousingData
+    {
+        Size = 600f,
+        HistoricalPrices = new float[] { 100000f ,125000f ,122000f },
+        CurrentPrice = 170000f
+    },
+    new HousingData
+    {
+        Size = 1000f,
+        HistoricalPrices = new float[] { 200000f, 250000f, 230000f },
+        CurrentPrice = 225000f
+    },
+    new HousingData
+    {
+        Size = 1000f,
+        HistoricalPrices = new float[] { 126000f, 130000f, 200000f },
+        CurrentPrice = 195000f
+    },
+    new HousingData
+    {
+        Size = 850f,
+        HistoricalPrices = new float[] { 150000f,175000f,210000f },
+        CurrentPrice = 205000f
+    },
+    new HousingData
+    {
+        Size = 900f,
+        HistoricalPrices = new float[] { 155000f, 190000f, 220000f },
+        CurrentPrice = 210000f
+    },
+    new HousingData
+    {
+        Size = 550f,
+        HistoricalPrices = new float[] { 99000f, 98000f, 130000f },
+        CurrentPrice = 180000f
+    }
+};
+```
+
+Use the [`TrainTestSplit`](xref:Microsoft.ML.DataOperationsCatalog.TrainTestSplit*) method to split the data into train and test sets. The result will be a [`TrainTestData`](xref:Microsoft.ML.DataOperationsCatalog.TrainTestData) object which contains two [`IDataView`](xref:Microsoft.ML.IDataView) members, one for the train set and the other for the test set. How much of the data is split is determined by the `testFraction` parameter. The snippet below is holding out 20 percent of the original data for the test set.
+
+```csharp
+DataOperationsCatalog.TrainTestData dataSplit = mlContext.Data.TrainTestSplit(data, testFraction: 0.2);
+IDataView trainData = dataSplit.TrainSet;
+IDataView testData = dataSplit.TestSet;
+```
+
+## Preapre the data
+
+Before using the data to train a machine learning model, it needs to be pre-processed. More information on data preparation can be found on the [data prep how-to article](prepare-data-ml-net.md) as well as the [`transforms page`](../resources/transforms.md).
+
+ML.NET applies certain defaults with regards to column types and names when building machine learning models .
+
+### Working with default column types
+
+By default, the input expected by the machine learning algorithms is a numerical vector. One way to specify that the input data will be a vector is in the data model with the [`VectorType`](xref:Microsoft.ML.Data.VectorTypeAttribute) attribute. This approach is appropriate when all of the data is already in numerical format and is intended to be ingested and processed together (i.e. image pixels). 
```

Comment:
How about something like: "The machine learning algorithms default input  is a numerical vector. Use the [`VectorType`](xref:Microsoft.ML.Data.VectorTypeAttribute) attribute for vector input data."?